### ISSUE TITLE
ColumnAttribute: add bool for primary key

### ISSUE THREAD

**Issue by penCsharpener** [sqlkata/querybuilder#246]
Love the new attributes for properties. 
A useful and cheap addition would be a second ctor where user can mark a property as being the primary key column. This could make update statements more straight forward as a where clause does have to be specified. 
It's maybe more tricky for composite primary keys.

**Comment by mota57** [comment]
Do you mean something like this? if you can provide more examples.


```
public class Loan {
  [Key]
  public string Id {get;set}

  public string Name {get; set;}
}
 var loanObj= new  Loan() { Id = 1, Name ="foo"}
 var query = db.Query("accounts").AsUpdate(loanObj);
 var compiled = db.Compiler.Compile(query);
Console.WriteLine(compiled) // UPDATE Loan SET Name = "foo" WHERE Id = 1
```

**Comment by mota57** [comment]
@ahmad-moussawi would you consider this feature for the library?

**Comment by penCsharpener** [comment]
@mota57 kind of, 
since attributes are retrieved through System.Reflection and SqlKata aims to be fast I would want to use as little as Reflection as possible. Hence I didn't want to add another attribute as in your example although that would be more neat.

```cs
public class Loan {
  [Column("Id", IsPrimary: true)]
  public string Id {get;set}

  public string Name {get; set;}
}
```

**Comment by ahmad-moussawi** [comment]
Nice feature, using reflection with caching is the ideal case here,
@mota57 would you be able to add this?

**Comment by penCsharpener** [comment]
https://github.com/henkmollema/Dapper-FluentMap/tree/master/src/Dapper.FluentMap
#245 is related. I suggest to look at how Dapper.FluentMap did it. It's easy to use.
I only recently started using sqlkata from source in my project so as soon as I'm more familiar with it I could work on an implementation myself. But that might be a few weeks till then.

**Comment by penCsharpener** [comment]
btw with my repo dapper-kata I wanted to build a light orm using both Dapper and SqlKata but since SqlKata.Execution already uses Dapper, I'd be happy investing that time into sqlkata instead.

**Comment by mota57** [comment]
Ok, will work on this. I just going to take sometime

**Comment by mota57** [comment]
@penCsharpener 
As you said that the attribute Key will be more "neat" take in consideration the following things:

1-  You will have to write twice the name of an attribute. For example, take a look at the property **Id**. 
```.cs
public class Loan 
{
  [Key("LoanId")]
  [Column("LoanId")]
  public string Id {get;set}
  
  [Key]
  public string OtherId {get; set;}

  public string Name {get; set;}
}
```


2- When you make an update it will generate a query like so, so it will also update the same keys which you may not like.

``` .sql
 "UPDATE [Loan] SET [LoanId] = 2, [OtherId] = 3, [Name] = 'faa'  WHERE [LoanId] = 2 AND [OtherId] = 3"

```

I still can go with the previous version to check if the column isPrimary, so what do think? Also, I have to work the cache.

**Comment by penCsharpener** [comment]
Since Column attribute already sounds very general why not make Key attribute inherite from Column attribute. This way you need to add only one attribute but can still give its SQL name. 
Since caching of attributes is coming a separate Key Attribute won't affect performance. 

If I'm not mistaken targeting .net standard  2.0 will give you more options with attribute handling.

**Comment by mota57** [comment]
Ok I agree it would be something like this
```.cs
using System;

namespace SqlKata
{
    /// <summary>
    /// This class is used as metadata on a property to generate different name in the output query.
    /// </summary>
    public class ColumnAttribute : Attribute
    {
        public string Name { get; private set; }
        public ColumnAttribute(string name)
        {
            if (string.IsNullOrEmpty(name))
            {
                throw new ArgumentNullException("Name parameter is required");
            }
            Name = name;
        }

    }


    /// <summary>
    /// This class is used as metadata on a property to generate different name in the output query.
    /// </summary>
    public class KeyAttribute : ColumnAttribute
    {
        public KeyAttribute([System.Runtime.CompilerServices.CallerMemberName] string name = "" ) 
        : base(name)
        {
            if (string.IsNullOrEmpty(name))
            {
                throw new ArgumentNullException("Name parameter is required");
            }
        }

    }

}
```

**Comment by mota57** [comment]
I update the comment because I notice the other methods they dont use this functionality because in the other methods you have to specify yoursel the columns and values. e.g

```
var cols = new [] {"Name", "Price"};

var data = new [] {
    new object[] { "A", 1000 },
    new object[] { "B", 2000 },
    new object[] { "C", 3000 },
};

db.Query("Products").Insert(cols, data);
```

I will push the branch later at night on my time.

**Comment by mota57** [comment]
@penCsharpener can you close the issue?

**Comment by penCsharpener** [comment]
Yes, I'm happy to close it but there isn't a nuget release yet with this. Would that be possible @ahmad-moussawi?

**Comment by rwood** [comment]
I had some issues when I ran this code.  There needed to be some changes made.  Here's my PR.
https://github.com/sqlkata/querybuilder/pull/258
